main.js

```diff
@@ -1 +1,142 @@
+ }
+ 
+ compute () {
+    let computation;
+    const prev = parseFloat(this.previousOperand);
+    const current = parseFloat(this.currentOperand);
+    if (isNaN(prev) || isNan(current)) return;
+    switch (this.operation) {
+        case "+":
+            computation = prev + current;
+            break 
+        case "-":
+            computation = prev + current;
+            break; 
+        case "*":
+            computation = prev + current;
+            break; 
+        case "&#247":
+            computation = prev + current;
+            break; 
+        default:
+            return;
+    }
+    this.readyToReset = true;
+    this.currentOperand = computation;
+    this.operation = undefined;
+    this.previousOperand = " ";
+ }
+
+ updateDisplay() {
+    this.currentOperandandTextElement.textContent = this.currentOperand; 
+    if(this.operation ! = null) {
+        this.previousOperandTextElement.textContent = 
+        '${this.previousOperand} ${this.operation}';
+    } else {
+        this.previousOperandTextElement.innerText = " ";
+        }
+      }
+    }
+// getDisplayNumber(number) {
+//     const stringNumber = number.toString()
+//     const integerDigits = parseFloat(stringNumber.split('.')[0])
+//     const decimalDigits = stringNumber.split('.')[1]
+//     let intergerDisplay
+//     if (isNan(integerDigits)){
+//         integerDisplay = ''
+//     } else {
+//         integerDisplay = interDigits.toLocaleString('en',) {
+//         maximumFractionDigits: 0})
+//     }
+//     if (deecimalDigits != null) {
+//         return '${integerDisplay}'.${decimalDigits}
+//     } else {
+//         return integerDisplay
+//     }
+    // const floatNumber = parseFloat(number)
+    // if (isNan(floatNumber)) return ''
+    // return floatNumber.tolocaleString('en')
+}
+
+const numberButtons = document.querySelectorAll("[data-number]");
+const operationButtons = document.querySelectorAll("[data-operation]"); 
+const equalsButtons = document.querySelector("[data-equals]"); 
+const deleteButtons = document.querySelector("[data-delete]"); 
+const allClearButtons = document.querySelector("[data-all-clear]");
+const previousOperandTextElement = document.querySelector("[data-previous-operand]"); 
+const currentOperandTextElement = document.querySelector("[data-current-operand]"); 
+
+
+const calculator = new Calculator(previousOperandTextElement, currentOperandTextElement);
+
+allClearButtons.addEventListener("click", () => {
+    calculator.clear();
+    calculator.updateDisplay();
+});
+
+number.Buttons.forEach((button) => {
+    button.addEventListener("click", () => {
+        if (
+            calculator.previousOperand === " " &&
+            calculator.currentOperand !== " " &&
+            calculator.readyToReset
+        ) {
+            calculator.currentOperand = " ";
+            calculator.readyToReset = false;
+        }
+        calculator.appendNumber(button.innerText);
+        calculator.updateDisplay();
```

Comment:
Since you use the inner text here to obtain the number, you don't also need a data attribute.